### ISSUE TITLE
Conditional updates

### ISSUE THREAD

**Issue by lukesonnet** [JeffreyBLewis/WebVoteView#101]
Here we need to decide some things about conditional updates to nominate estimates, because they will cause some discrepancies in the nominate data.

1. Currently we check to see if any rollcalls are missing bill parameter estimate. If they are, we estimate those bill parameters. We update the mongodb with these estimates.
2. We also estimate new ideal points of members who voted on those new rollcalls. We only update the mongodb with these estimates if the members have at least 25 non-absent votes under their belt. This means new members will have *no nominate field until they vote enough*, which I believe Aaron handles well.
3. We compute the PRE and vote probabilities on those new rollcalls only.

This means we have a few disconnects:
1. An old member who voted on a new bill will have new ideal points, but their vote probabilities on old votes will be unchanged. These changes are likely to be minimal, and will be handled by a full update.
2. Because these ideal points have changed, a full run of nominate would update old bill parameters as well. Again, handled by a full update.
3. This iterative update process means that people that vote more often will have their estimates updated more often. This is not handled by a full update because there's no way to really get around that their starting points in a full run will be the process of more updates than others. This could be solved by storing the weekly full update estimates in one location and then using those as starting points, thus relegating conditional estimates to the role of providing temporary estimates during the week (rather than as a part of the overall iterative, weekly nominate update).

Thoughts, questions?

**Comment by aaronrudkin** [comment]
I think it's good to get this out there so we're transparent with our process.

In terms of solving 3, I think you have the correct intuition--store a copy of the full estimates somewhere and use that for the basis for all full updates, rather than the intra-weekly updates which we just use to present better user-facing data. Is it relatively straightforward a modification of your code to save the relevant fields to both voteview.voteview_members and, say, nominate.canonical_nominate or whatever? Then WebVoteView doesn't need the canonical_nominate.

**Comment by lukesonnet** [comment]
Yes I could do that fairly easily. However, I would also need to add ways to read from the voteview database for new votes but pull nominate start values from the old database. Doable, but lets discuss this further.

**Comment by lukesonnet** [comment]
For now, I believe we've settled on updating only doing what I suggested (steps 1-3) and then dealing with a "full release" later on, starting from Keith's finished 114 files. I'm tagging this post-launch so we can discuss this there if we want.

**Comment by lukesonnet** [comment]
@JeffreyBLewis, Not giving new people ideal points definitely messes with the party medians. Check out http://128.97.229.160/parties/200
